Don't rely on odd index

gonuke opened the pull request:
This suggestion makes a number of changes.  In general, it is a little
more verbose with variables for long term clarity.  Specifically, it no
longer relies on the counting the entries in the token_list and acting
on the odd ones defined by (i & 1 == 1).  Instead, it pops tokens off
the list and acts on them according to their characteristics.  Some
passes through the while-loop only consume a single token from
the list while others consume 2 tokens from the list.

gonuke committed:
Don't rely on odd index

This suggestion makes a number of changes.  In general, it is a little
more verbose with variables for long term clarity.  Specifically, it no
longer relies on the counting the entries in the token_list and acting
on the odd ones defined by (i & 1 == 1).  Instead, it pops tokens off
the list and acts on them according to their characteristics.  Some
passes through the while-loop only consume a single token from
the list while others consume 2 tokens from the list.

gonuke committed:
initialize defaul table_ids entry as dict

pep8speaks commented:
Hello @gonuke! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`tests/test_mcnp.py`](https://github.com/mehmeturkmen/pyne/blob/b1eedc46d4f2cb4e4081248e3f72a4f68555c2cf/tests/test_mcnp.py):

> [Line 515:13](https://github.com/mehmeturkmen/pyne/blob/b1eedc46d4f2cb4e4081248e3f72a4f68555c2cf/tests/test_mcnp.py#L515): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent
> [Line 516:13](https://github.com/mehmeturkmen/pyne/blob/b1eedc46d4f2cb4e4081248e3f72a4f68555c2cf/tests/test_mcnp.py#L516): [E128](https://duckduckgo.com/?q=pep8%20E128) continuation line under-indented for visual indent



##### Comment last updated at 2020-08-17 01:22:52 UTC

gonuke committed:
cleanup for PEP8

gonuke committed:
Use existing scheme

Apparently, there is already a scheme for storing and writing this
data from a Material object.

gonuke committed:
update test for prior scheme

gonuke committed:
get correct size for array of strings

gonuke committed:
don't get ahead of object creation

gonuke committed:
updated test to match old scheme

gonuke committed:
PEP8

gonuke committed:
manage case variations

gonuke committed:
tweak indenting (ignoring PEP8 for now)

gonuke committed:
PEP8 except indenting...
